src/Symfony/Component/Process/PhpSpecifExecutableFinder.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Process;
+
+use Symfony\Component\Process\Exception\PhpSpecifExecutableInvalidVersionException;
+use Symfony\Component\Process\Exception\PhpSpecifExecutableNotFoundException;
+
+/**
+ * An executable finder specifically designed for the PHP executable.
+ * This class allows searching for a specific version of PHP if provided.
+ *
+ * @author Pululu Kinaga Andre <pululuandre@gmail.com>
+ */
+class PhpSpecifExecutableFinder
+{
+    /**
+     * Finds the PHP executable path, optionally for a specific version.
+     *
+     * @return string The path to the PHP executable.
+     * @throws PhpSpecifExecutableNotFoundException If no executable is found.
```

Comment:
I suggest returning `null` instead of throwing for that case, as it is not that exceptional that a version is not found on the system.

Note that I suggest `null`, not `false` as done in the ExecutableFinder, as it is much cleaner as a way to represent a non-existent result.